File: 11_file_backup_system/cpp/file_backup_system.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main() {
    string source, backup;

    cout << "Enter source file: ";
    cin >> source;

    cout << "Enter backup file: ";
    cin >> backup;

    ifstream in(source, ios::binary);
    ofstream out(backup, ios::binary);

    if (!in) {
        cout << "Source file not found.\n";
        return 0;
    }

    char ch;
    while (in.get(ch)) {
        out.put(ch);
    }

    in.close();
    out.close();

    cout << "Backup created successfully.\n";
    return 0;
}

File: 11_file_backup_system/python/file_backup_system.py
source = input("Enter source file: ")
backup = input("Enter backup file: ")

try:
    with open(source, "rb") as src:
        data = src.read()

    with open(backup, "wb") as dst:
        dst.write(data)

    print("Backup created successfully.")
except FileNotFoundError:
    print("Source file not found.")
